### ISSUE TITLE
helm lint of packaged charts (tgz) not finding Chart.yaml with some SemVer2 compliant version values.

### ISSUE THREAD

**Issue by dvpierce** [helm/helm#6636]
If I set the "version" field in my Chart.yaml to a version with a pre-release value (e.g., 1.0.1-development) the helm lint of the directory will succeed, but the helm lint of the packaged tgz file (e.g., davesproject-1.0.1-development.tgz) will fail.

Replacing the hyphen in the version name with another permitted character (e.g., 1.0.1+development) and repacking (davesproject-1.0.1+development.tgz) works fine as a workaround, but I'm assuming this wasn't intentional.

Thanks,
Dave P.

```
[pierdavpa@lin2dv2ap55 _helm]$ helm lint davetestanother-1.0.4-development.tgz
==> Skipping davetestanother-1.0.4-development.tgz
No chart found for linting (missing Chart.yaml)

Error: 0 chart(s) linted, 1 chart(s) failed
[pierdavpa@lin2dv2ap55 _helm]$ helm lint davetestanother-1.0.4+development.tgz
==> Linting davetestanother-1.0.4+development.tgz
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, no failures
[pierdavpa@lin2dv2ap55 _helm]$
```

Output of `helm version`:
`Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}`

Output of `kubectl version`:
`Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:54Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}`

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  AKS/k8s hybrid

**Comment by kroonprins** [comment]
Similar error here for "-rc1" version: https://github.com/kroonprins/helm-lint-error

**Comment by kroonprins** [comment]
Was fixed in helm 2.15 by this commit https://github.com/helm/helm/commit/bfc0d76fff148865aa1e9d5ba28693da2947b4fe
